<?xml version="1.0"?>
<sdf version="1.7">
    <!--include defination-->
    <xacro_include_definition uri="model://ur5/ur5.def.xacro" />
    <xacro_include_definition uri="model://ur5/ur5_position_controller.def.xacro" />
    <!--model-->
    <model name="ur5">
        <xacro_macro name="ur5" parent="__model__"/>
        <!--plugins for ur-->
        <xacro_macro name="ur5_position_controller"/>
        <!--joint state publisher-->
        <plugin filename="ignition-gazebo-joint-state-publisher-system" 
                name="ignition::gazebo::systems::JointStatePublisher">
            <joint_name>shoulder_pan_joint</joint_name>
            <joint_name>shoulder_lift_joint</joint_name>
            <joint_name>elbow_joint</joint_name>
            <joint_name>wrist_1_joint</joint_name>
            <joint_name>wrist_2_joint</joint_name>
            <joint_name>wrist_3_joint</joint_name>
        </plugin>
    </model>
</sdf>